enhancements/installer/packet-ipi.md

```diff
@@ -0,0 +1,294 @@
+
+  A generic RHCOS Ignition URL could instruct nodes booted on Packet to
+  configure their network using the EC2-like [metadata
+  service](https://www.packet.com/developers/docs/servers/key-features/metadata/).
+
+- Packet does not offer a managed DNS solution. Route53, CloudFlare, or RFC 2136
+  (DNS Update), and NS record based zone forwarding are considerations. Some
+  existing IPI drivers ([such as
+  `baremetal`](https://github.com/openshift/installer/blob/master/docs/user/metal/install_ipi.md))
+  have similar limitations.
+- Packet does not offer a loadbalancer service, but does [offer the networking
+  features](https://www.packet.com/developers/docs/network/) that makes this
+  possible at the device level.
+- [Packet CCM currently offers
+  LoadBalancing](https://github.com/packethost/packet-ccm#load-balancers) by
+  creating and maintaining a MetalLB deployment within the cluster.  This may be
+  impacted by <https://github.com/openshift/enhancements/pull/356>.
+- [Packet
+  CSI](https://github.com/packethost/csi-packet#kubernetes-container-storage-interface-csi-plugin-for-packet)
+  currently requires a minimum of 100GB per volume. Additionally, the [Packet
+  EBS service](https://www.packet.com/developers/docs/storage/ebs/) is available
+  in a limiting subset of the available facilities.
+- Both Public and Private IPv4 and IPv6 addresses made available for each node.
+  Private and VLAN Networks can be restricted at the project level (default) or
+  node by node. While custom ASNs and BGP routing features are supported, these
+  and other networking features will not be exposed through the IPI.
+  <https://www.packet.com/cloud/network/>
+- Openshift will run on RHCOS directly on bare metal without the need for a
+  VM layer.
+
+### Risks and Mitigations
+
+The [official Packet
+ClusterAPI](https://github.com/kubernetes-sigs/cluster-api-provider-packet),
+[CCM](https://github.com/packethost/packet-ccm), and
+[CSI](https://github.com/packethost/csi-packet) drivers are all relatively new
+and may make undertake substantial design changes.  This project will either
+need to adopt and maintain the current releases, or adopt newer releases
+developed alongside this project.
+
+The CI will need to assure that Packet resources are cleanly released after
+failed and successful builds.  Packet is currently used in the bare metal
+provisioning tests, so there is some prior art available.  If the same account
+is used for testing the IPI, it may need account quota increases.
+
+## Design Details
+
+### Open Questions
+
+- Is the Packet CCM necessary?
+  - If so, who is responsible for supporting the MetalLB installation the Packet
+    CCM provides and requires for LoadBalancer definitions?
+  - Can [https://github.com/plunder-app/kube-vip#56](plunder-app/kube-vip)
+    stand-in for LoadBalancer service on Packet while disabling the packet-ccm's
+    MetalLB approach to LoadBalancing?
```

Comment:
I don't see kube-vip and MetalLB as doing quite the same thing, since MetalLB handles Services of type=LoadBalancer, and kube-vip is focused on being the load balancer for the control plane.  Right now I do not think it's likely we would adopt kube-vip.  What is more likely is evolving OpenShift to where we can use LoadBalancer Services for our control plane.  See #459 for the latest discussion on this topic.  You may want to link that enhancement here for reference.,

Reply:
https://kube-vip.io/
> A Load-Balancer for both inside and outside a Kubernetes cluster
> `type: LoadBalancer`

I'm adding a note about #459.